static/src/javascripts/lib/storage.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+/* eslint consistent-return: 0 */
+
+class Storage {
+    storage: any;
+    available: ?boolean;
+
+    constructor(type: string) {
+        this.storage = window[type];
+        this.available = this.isAvailable();
+    }
+
+    isAvailable(): ?boolean {
+        const key = 'local-storage-module-test';
+
+        if (this.available !== undefined) {
+            return this.available;
+        }
+
+        try {
+            // to fully test, need to set item
+            // http://stackoverflow.com/questions/9077101/iphone-localstorage-quota-exceeded-err-issue#answer-12976988
+            this.storage.setItem(key, 'graun');
+            this.storage.removeItem(key);
+            this.available = true;
+        } catch (err) {
+            this.available = false;
+        }
```

Comment:
I feel that this try / catch block belongs in the constructor. Then `isAvailable` will become a simple getter method. What do you think?